include/seastar/coroutine/exception.hh

```diff
@@ -87,9 +93,10 @@ struct exception {
 /// co_return coroutine::make_exception(std::runtime_error("something failed miserably"));
 /// ```
 template<typename T>
+requires std::same_as<T, std::exception_ptr> || std::derived_from<std::remove_reference_t<T>, std::exception>
```

Comment:
This is just an indication the API is wrong. It behaves differently for different argument types. It should be two separate APIs.

Reply:
Agreed. Done that way in v3.